ibverbs/src/lib.rs

```diff
@@ -1197,6 +1190,48 @@ impl<'ctx> ProtectionDomain<'ctx> {
             Ok(MemoryRegion { mr, data })
         }
     }
+
+    /// Allocates and registers a Memory Region (MR) associated with this `ProtectionDomain`.
+    ///
+    /// This process allows the RDMA device to read and write data to the allocated memory. Only
+    /// registered memory can be sent from and received to by `QueuePair`s. Performing this
+    /// registration takes some time, so performing memory registration isn't recommended in the
+    /// data path, when fast response is required.
+    ///
+    /// Every successful registration will result with a MR which has unique (within a specific
+    /// RDMA device) `lkey` and `rkey` values. These keys must be communicated to the other end's
+    /// `QueuePair` for direct memory access.
+    ///
+    /// The maximum size of the block that can be registered is limited to
+    /// `device_attr.max_mr_size`. There isn't any way to know what is the total size of memory
+    /// that can be registered for a specific device.
+    ///
+    /// `allocate` currently sets the following permissions for each new `MemoryRegion`:
```

Comment:
Let's also add a line after the list that point at `allocate_with_permissions` if users need more control :+1: